internal/pipe/brew/brew.go

```diff
@@ -222,12 +216,27 @@ func dataFor(ctx *context.Context, cfg config.Homebrew, artifacts []artifact.Art
 		}
 
 		if cfg.URLTemplate == "" {
-			cfg.URLTemplate = fmt.Sprintf(
-				"%s/%s/%s/releases/download/{{ .Tag }}/{{ .ArtifactName }}",
-				ctx.Config.GitHubURLs.Download,
-				ctx.Config.Release.GitHub.Owner,
-				ctx.Config.Release.GitHub.Name,
-			)
+			switch tokenType {
+			case context.TokenTypeGitHub:
+				// https://github.com/goreleaser/goreleaser/releases/download/v0.112.1/goreleaser_Linux_arm64.tar.gz
+				cfg.URLTemplate = fmt.Sprintf(
+					"%s/%s/%s/releases/download/{{ .Tag }}/{{ .ArtifactName }}",
+					ctx.Config.GitHubURLs.Download,
+					ctx.Config.Release.GitHub.Owner,
+					ctx.Config.Release.GitHub.Name,
+				)
+			case context.TokenTypeGitLab:
+				//https://gitlab.com/mavogel/release-testing/uploads/22e8b1508b0f28433b94754a5ea2f4aa/release-testing_0.3.7_Darwin_x86_64.tar.gz
+				cfg.URLTemplate = fmt.Sprintf(
+					"%s/%s/%s/releases/uploads/%s/{{ .ArtifactName }}",
+					ctx.Config.GitLabURLs.Download,
+					ctx.Config.Release.GitLab.Owner,
+					ctx.Config.Release.GitLab.Name,
+					artifact.Extra["GitLabFileUploadHash"],
```

Comment:
maybe use https://godoc.org/github.com/goreleaser/goreleaser/internal/artifact#Artifact.ExtraOr with some fake string?

in theory it will never happen, but will be easier to debug in case it does (better have a url with `fake-hash-should-not-happen` than just `nil` 🤔

or maybe handle it and fail earlier